src/main/java/org/apache/mesos/scheduler/repair/CoreFrameworkState.java

```diff
@@ -0,0 +1,44 @@
+package org.apache.mesos.scheduler.repair;
+
+import org.apache.mesos.Protos.FrameworkID;
+import org.apache.mesos.Protos.TaskID;
+import org.apache.mesos.Protos.TaskInfo;
+import org.apache.mesos.state.StateStoreException;
+
+import java.util.List;
+
+/**
+ * Exposes simple task-oriented API for components that need to interact with the framework
+ * as a whole.
+ */
+public interface CoreFrameworkState {
```

Comment:
The RepairScheduler needs to do something with stopped tasks. The way I implemented it is that stopped tasks are removed from the StateStore, which enables them to be replaced. Would you like me to use the `clearTask()` method here? It would have the same effect, except it may result in untracked orphaned tasks during partitions and potential races due to using the name rather than the ID. If you're okay with those shortcomings, I will change the implementation.